lighthouse-core/lib/network-request.js

```diff
@@ -125,27 +126,31 @@ module.exports = class NetworkRequest {
   onRequestWillBeSent(data) {
     this.requestId = data.requestId;
 
-    const url = new URL(data.request.url);
-    this.url = data.request.url;
-    this.documentURL = data.documentURL;
-    this.parsedURL = {
-      scheme: url.protocol.split(':')[0],
-      // Intentional, DevTools uses different terminology
-      host: url.hostname,
-      securityOrigin: url.origin,
-    };
-    this.isSecure = SECURE_SCHEMES.includes(this.parsedURL.scheme);
-
-    this.startTime = data.timestamp;
-
-    this.requestMethod = data.request.method;
-
-    this.initiator = data.initiator;
-    this.resourceType = data.type && RESOURCE_TYPES[data.type];
-    this.priority = data.request.initialPriority;
-
-    this.frameId = data.frameId;
-    this.isLinkPreload = data.initiator.type === 'preload' || !!data.request.isLinkPreload;
+    try {
+      // try to construct the url and fill in request
```

Comment:
Should we unconditionally catch all errors in here? Admittedly there's not anything else in here that will throw if our type checking is correct, but we might add more functionality in the future and any errors will fail silently.

What do you think about limiting the try/catch to the url constructor?

Like
```js
let url;
try {
  // try to construct the url and fill in request
  url = new URL(data.request.url);
} catch (e) {
  return; // isValid left false, all other data is blank
}
this.isValid = true;
this.url = data.request.url;
// ...
```

Reply:
Yup that was my first instinct!  Reverted.